synthpop/ipu/ipu.py

```diff
@@ -261,9 +263,33 @@ def household_weights(
         iterations += 1
 
         if iterations > max_iterations:
-            raise RuntimeError(
-                'Maximum number of iterations reached during IPU: {}'.format(
-                    max_iterations))
+            if ignore_max_iters:
+                fitting_tolerance = fit_change - convergence
+                print('Fitting tolerance before 20000 iterations: %s' % str(fitting_tolerance))
+                ipu_dict = {'best_fit_qual': best_fit_qual,
+                            'fit_change': fit_change,
+                            'fitting_tolerance': fitting_tolerance,
+                            'geog_id': geography}
+                if isinstance(geography, pd.Series):
+                    state, county = geography['state'], geography['county']
+                    tract, bgroup = geography['tract'], geography['block group']
+                    np.save('max_iter_{}_{}_{}_{}.npy'.format(state, county,
+                                                              tract, bgroup), ipu_dict)
+                elif isinstance(geography, list):
+                    np.save('max_iter_{}_{}.npy'.format(geography[0], geography[1]), ipu_dict)
+                else:
+                    np.save('max_iter_{}.npy'.format(str(geography)), ipu_dict)
```

Comment:
@msoltadeo , I've been doubting regarding this feature becasue I realized that if we add a `year` paramter to `synthesize` first and to `synthesize_all` secondly we will be changing the way these methods work just to give a name to a file in `household_weights`. And since it is something optional (users can decide to keep the default False for ignore_max_iters) we will need to add a new parameter when in some cases this won't be used. So, adapting all relative changes to this new parameter (unit tests, etc.) to a parameter that wont possibly be used is not convincing me. What do you think?